api/src/main/java/org/pmiops/workbench/api/CohortController.java

```diff
@@ -1,29 +1,174 @@
 package org.pmiops.workbench.api;
 
-import com.google.common.collect.ImmutableList;
-import java.util.List;
+import java.sql.Timestamp;
+import java.time.Clock;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+import javax.inject.Provider;
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+import org.pmiops.workbench.db.dao.CohortDao;
+import org.pmiops.workbench.db.dao.WorkspaceDao;
+import org.pmiops.workbench.db.model.User;
+import org.pmiops.workbench.db.model.Workspace;
+import org.pmiops.workbench.exceptions.NotFoundException;
 import org.pmiops.workbench.model.Cohort;
+import org.pmiops.workbench.model.CohortListResponse;
+import org.pmiops.workbench.model.DataAccessLevel;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.RestController;
 
 @RestController
-public class CohortController implements CohortsApi {
+public class CohortController implements CohortsApiDelegate {
+
+  private static final Logger log = Logger.getLogger(CohortController.class.getName());
+
+  private static final Function<org.pmiops.workbench.db.model.Cohort, Cohort> TO_CLIENT_COHORT =
```

Comment:
Added a comment.

The client representation is generated by Swagger. The backend representation (db.model.Cohort) is written by hand with annotations.

I toyed with the notion of instead just using the Swagger generated code and using XML configuration with it, but in practice it won't work well -- the client representation doesn't perfectly match what we want for the DB layer. (JODA is one good example.)